static/js/components/GroupPageManageUserButtons.jsx

```diff
@@ -63,49 +76,88 @@ const ManageUserButtons = ({ group, loadedId, user, isAdmin }) => {
-        size="small"
-        onClick={() => {
-          toggleUserAdmin(user);
-        }}
-        disabled={isAdmin(user) && numAdmins === 1}
-      >
-        <span style={{ whiteSpace: "nowrap" }}>
-          {isAdmin(user) ? "Revoke admin status" : "Grant admin status"}
-        </span>
-      </Button>
-      &nbsp;|&nbsp;
-      <Tooltip title="Manage whether user can save sources to this group.">
-        <Button
-          size="small"
-          onClick={() => {
-            toggleUserCanSave(user);
-          }}
-        >
-          <span style={{ whiteSpace: "nowrap" }}>
-            {canSave(user) ? "Revoke save access" : "Grant save access"}
-          </span>
-        </Button>
-      </Tooltip>
-      &nbsp;|&nbsp;
-      <IconButton
-        edge="end"
-        aria-label="delete"
-        data-testid={`delete-${user.username}`}
-        onClick={() =>
-          dispatch(
-            groupsActions.deleteGroupUser({
-              userID: user.id,
-              group_id: group.id,
-            })
-          )
-        }
-        disabled={isAdmin(user) && numAdmins === 1}
-      >
-        <DeleteIcon />
-      </IconButton>
+      {isAdmin(currentUser) && (
+        <div>
+          <Button
+            size="small"
+            onClick={() => {
+              toggleUserAdmin(user);
+            }}
+            disabled={isAdmin(user) && numAdmins === 1}
+          >
+            <span style={{ whiteSpace: "nowrap" }}>
+              {isAdmin(user) ? "Revoke admin status" : "Grant admin status"}
+            </span>
+          </Button>
+          &nbsp;|&nbsp;
+          <Tooltip title="Manage whether user can save sources to this group.">
+            <Button
+              size="small"
+              onClick={() => {
+                toggleUserCanSave(user);
+              }}
+            >
+              <span style={{ whiteSpace: "nowrap" }}>
+                {canSave(user) ? "Revoke save access" : "Grant save access"}
+              </span>
+            </Button>
+          </Tooltip>
+          &nbsp;|&nbsp;
+        </div>
+      )}
+      {(isAdmin(currentUser) || user.username === currentUser.username) && (
+        <div>
+          <IconButton
+            edge="end"
+            aria-label="delete"
+            data-testid={`delete-${user.username}`}
+            onClick={() => setConfirmDeleteOpen(true)}
+            disabled={isAdmin(user) && numAdmins === 1}
+          >
+            <DeleteIcon />
+          </IconButton>
+          <Dialog
+            fullWidth
+            open={confirmDeleteOpen}
+            onClose={handleConfirmDeleteDialogClose}
+          >
+            <DialogTitle>Remove user?</DialogTitle>
+            <DialogContent dividers>
+              <DialogContentText>
+                Are you sure you want to delete this user from this group?
+                <br />
+                Warning! This will delete the user from the group and all of its
+                filters.
```

Comment:
Should have a different title user:
  1) "yourself" and "you" if it's the current user
  2) Otherwise "this user" and "the user"